Add request types to TS definitions

bhelx opened the pull request:
Previously, the `body` parameter of the operations was just defined as a plain `object`. This uses their proper Request types as we do in the other statically typed clients. We're not considering this a breaking change because if you had passed an invalid object before, the server would already be stopping you. We are to other interpretations from our typescript users.

Example use:

```js

import * as recurly from "recurly";

const client = new recurly.Client('<apikey>')

async function runIt() {

  const acctReq: recurly.AccountCreate = {
    code: 'test12345678',
    firstName: 'Benjamin',
    lastName: 'Du Monde'
  }
  let account = await client.createAccount(acctReq)
  console.log(account)
  account = await client.getAccount(account.id)
  console.log(account)
  // plain objects still work as long as they conform to schema
  const updates = {
    firstName: 'Bill',
    lastName: 'Cauchois'
  }
  account = await client.updateAccount(account.id, updates)
  console.log(account)
  account = await client.deactivateAccount(account.id)
  console.log(account)
}

runIt()
```

bhelx commented on lib/recurly.d.ts:
```diff
@@ -8,2381 +8,2381 @@ export interface Site {
    * Site ID
    */
-  id: string | null;
+  id?: string | null;
```
I had to make all these optional so that they could be constructed properly:

![Screen Shot 2020-07-14 at 12 53 15 PM](https://user-images.githubusercontent.com/185919/87459956-b55f9b80-c5d1-11ea-9674-752a792ba611.png)

@dbrudner does this seem correct?

dbrudner replied:
Yeah definitely, this is a great QOL improvement in general, and especially for TS apps with [strictNullChecks] (https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#--strictnullchecks) enabled.

The type lgtm. This lets users pass in either a string, explicit `null`, or leave off the parameter completely. Assuming the runtime allows for those three possible values, seems solid.

bhelx committed:
Support request types as bodies

Previously, the body parameter of the operations was just defined as a plain object.
This uses their proper Request types as we do in the other statically typed clients.
We're not considering this a breaking change because if you had passed an invalid object before, the server would already be stopping you

douglasmiller reviewed (approved):
👍
